src/parser.rs

```diff
@@ -556,15 +563,15 @@ pub struct ScriptDataDate {
 #[allow(non_upper_case_globals)]
 static script_data_name_tag: &[u8] = &[2];
 
-pub fn script_data(input: &[u8]) -> IResult<&[u8], ScriptData<'_>> {
+pub fn script_data(input: &[u8]) -> IResult<&[u8], ScriptData> {
   // Must start with a string, i.e. 2
-  let (i, _) = tag(script_data_name_tag)(input)?;
-  let (i, (name, arguments)) = pair(script_data_string, script_data_value)(i)?;
-  Ok((i, ScriptData { name, arguments }))
+  tag(script_data_name_tag)(input)
+    .and_then(|(i, _)| pair(script_data_string, script_data_value)(i))
```

Comment:
Why not a single tuple for the 3 components?

Reply:
Yep, I will change that, thanks!